src/sentry/api/event_search.py

```diff
@@ -387,6 +387,20 @@ class SearchConfig:
     # Enables boolean filtering (AND / OR)
     allow_boolean = True
 
+    # Allows us to specify an allowlist of keys we will accept for this search.
+    # If empty, allow all keys.
+    allowed_keys: Set[str] = field(default_factory=set)
+
+    # Which key we should return any free text under
+    free_text_key = "message"
+
+    @classmethod
+    def create_from(cls, search_config: "SearchConfig", **overrides):
```

Comment:
Should this be a `@staticmethod` if you're not going to use the `cls`?

Alternatively swap out `SearchConfig` for `cls`